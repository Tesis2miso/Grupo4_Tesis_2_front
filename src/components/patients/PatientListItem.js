import * as React from 'react';
import List from '@mui/material/List';
import ListItem from '@mui/material/ListItem';
import Divider from '@mui/material/Divider';
import ListItemText from '@mui/material/ListItemText';
import ListItemAvatar from '@mui/material/ListItemAvatar';
import Avatar from '@mui/material/Avatar';
import Typography from '@mui/material/Typography';
import ListItemButton from '@mui/material/ListItemButton';
import { red, orange, blue, green, purple, pink } from '@mui/material/colors';
import { useNavigate } from 'react-router-dom';

function PatientListItem(props) {
  const randomColor = () => {
    const classes = [
      red, orange, blue, green, purple, pink
    ]
    const func = classes[Math.floor(Math.random() * classes.length)];
    return func[500];
  }
  const { user } = props;
  const navigate = useNavigate();
  const onSelected = () => {
    navigate(`/patients/${user.id}`)
  }

  return (
    <ListItem alignItems="flex-start">
      <ListItemButton onClick={onSelected}>
        <ListItemAvatar>
          <Avatar alt={user.name} src="/static/images/avatar/1.jpg" sx={{ bgcolor: randomColor() }} />
        </ListItemAvatar>
        <ListItemText
          primary={user.name}
          secondary={
            <React.Fragment>
              <Typography
                sx={{ display: 'inline' }}
                component="span"
                variant="body2"
                color="text.primary"
              >
                {user.email}
              </Typography>
              {` - ${user.phone}`}
            </React.Fragment>
          }
        />
      </ListItemButton>
    </ListItem>
  );
}

export default PatientListItem;
