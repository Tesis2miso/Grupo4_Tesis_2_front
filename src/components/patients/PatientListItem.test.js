import '@testing-library/jest-dom';
import { mount, render, shallow } from 'enzyme';
import axios from "axios";
import PatientListItem from './PatientListItem';

const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));
jest.mock('axios', () => jest.fn());
jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: (str) => str,
      i18n: {
        changeLanguage: () => new Promise(() => { }),
      },
    };
  },
  initReactI18next: {
    type: '3rdParty',
    init: () => { },
  }
}));

describe('Tests_patient_list_item', () => {

  var wrapper = null;
  var user = null;

  beforeEach(() => {
    user = {
      "birth_day": "1997-04-19T00:00:00",
      "city": "Bogot√°",
      "created_at": "2023-02-13T16:33:18.231963",
      "email": "drummerwilliam@gmail.com",
      "id": 1.0,
      "name": "William",
      "phone": "3013016284",
      "token": null
    }
    wrapper = shallow(<PatientListItem user={user} />);
  });

  test('show patient list item', () => {
    expect(wrapper).toMatchSnapshot();
  })

  test('go to patient detail', async () => {
    wrapper.find('.patientListItem').simulate('click', {
      preventDefault: () => { }
    });
  })
})