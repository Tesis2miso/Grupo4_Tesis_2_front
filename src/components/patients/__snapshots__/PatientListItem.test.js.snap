// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests_patient_list_item show patient list item 1`] = `
<ForwardRef(ListItem)
  alignItems="flex-start"
>
  <ForwardRef(ListItemButton)
    className="patientListItem"
    onClick={[Function]}
  >
    <ForwardRef(ListItemAvatar)>
      <ForwardRef(Avatar)
        alt="William"
        src="/static/images/avatar/1.jpg"
        sx={
          Object {
            "bgcolor": "#ff9800",
          }
        }
      />
    </ForwardRef(ListItemAvatar)>
    <ForwardRef(ListItemText)
      primary="William"
      secondary={
        <React.Fragment>
          <ForwardRef(Typography)
            color="text.primary"
            component="span"
            sx={
              Object {
                "display": "inline",
              }
            }
            variant="body2"
          >
            drummerwilliam@gmail.com
          </ForwardRef(Typography)>
           - 3013016284
        </React.Fragment>
      }
    />
  </ForwardRef(ListItemButton)>
</ForwardRef(ListItem)>
`;
